pkg/rbac/sharedData.go

```diff
@@ -17,6 +17,7 @@ import (
 type SharedData struct {
 	// These are the data fields.
 	csResources     []Resource // Cluster-scoped resources (ie. Node, ManagedCluster)
+	csResourcesMap  map[Resource]struct{}
```

Comment:
Wonder if we should avoid duplication and remove `csResources` and only use `csResourceMap` in all places.

Reply:
Will move in next PR